projects/plugins/jetpack/3rd-party/jetpack-backup.php

```diff
@@ -0,0 +1,132 @@
+
+/**
+ * Verify the intent to install Jetpack Backup, and kick off installation.
+ *
+ * This works in tandem with a JITM set up in the JITM package.
+ */
+function try_install() {
+	if ( ! isset( $_GET['jetpack-backup-action'] ) ) {
+		return;
+	}
+
+	check_admin_referer( 'jetpack-backup-install' );
+
+	$result = false;
+	// If the plugin install fails, redirect to plugin install page pre-populated with jetpack-backup search term.
+	$redirect_on_error = admin_url( 'plugin-install.php?s=jetpack-backup&tab=search&type=term' );
+
+	// Attempt to install and activate the plugin.
+	if ( current_user_can( 'activate_plugins' ) ) {
+		switch ( $_GET['jetpack-backup-action'] ) {
+			case 'install':
+				$result = install_and_activate();
+				break;
+			case 'activate':
+				$result = activate();
+				break;
+		}
+	}
+
+	$jetpack = \Jetpack::init();
+	if ( $result ) {
+		/** This action is already documented in _inc/lib/class.core-rest-api-endpoints.php */
+		do_action( 'jetpack_activated_plugin', PLUGIN_FILE, 'jitm' );
+		$jetpack->stat( 'jitm-plugin-install', 'jetpack-backup-success' );
+		$redirect = admin_url( 'admin.php?page=jetpack-backup' );
+	} else {
+		$jetpack->stat( 'jitm-plugin-install', 'jetpack-backup-fail' );
+		$redirect = add_query_arg( 'jetpack-backup-install-error', true, $redirect_on_error );
+	}
+
+	wp_safe_redirect( $redirect );
+
+	exit;
+}
+
+/**
+ * Install and activate the Jetpack Backup plugin.
+ *
+ * @return bool result of installation
+ */
+function install_and_activate() {
+	jetpack_require_lib( 'plugins' );
+	$result = \Jetpack_Plugins::install_and_activate_plugin( PLUGIN_SLUG );
+
+	if ( is_wp_error( $result ) ) {
+		return false;
+	} else {
+		return true;
+	}
+}
+
+/**
+ * Activate the Jetpack Backup plugin.
+ *
+ * @return bool result of activation
+ */
+function activate() {
+	$result = activate_plugin( PLUGIN_FILE );
+
+	// Activate_plugin() returns null on success.
+	return is_null( $result );
+}
+
+/**
+ * Notify the user that the installation of Jetpack Backup failed.
+ */
+function error_notice() {
+	if ( empty( $_GET['jetpack-backup-install-error'] ) ) { // phpcs:ignore WordPress.Security.NonceVerification.Recommended
+		return;
+	}
+
+	?>
+	<div class="notice notice-error is-dismissible">
+		<p><?php esc_html_e( 'There was an error installing Jetpack Backup. Please try again.', 'jetpack' ); ?></p>
+	</div>
+	<?php
+}
+
+/**
+ * Set some options when first activating the plugin via Jetpack.
+ *
+ * @since 8.9.0
+ *
+ * @param string $plugin_file Plugin file.
+ * @param string $source      Where did the plugin installation originate.
+ */
+function configure_plugin( $plugin_file, $source ) {
+	if ( PLUGIN_FILE !== $plugin_file ) {
+		return;
+	}
+
+	$plugin_info = array(
+		'plugin'  => 'jetpack',
+		'version' => JETPACK__VERSION,
+		'time'    => time(),
+		'source'  => esc_attr( $source ),
+	);
+
+	update_option( 'jetpack_backup_referred_by', $plugin_info );
```

Comment:
Since the Jetpack Backup plugin uses `Jetpack_Options`, maybe we could store the option there, for easier cleanup if the plugin eventually gets deleted? We could not do that with third-party plugins like Creative Mail, but here maybe that would be an option?

Reply:
Do you think we even need this option stored at all? I'm not even sure what it's used for.

Reply:
Might be useful and won't hurt to add. Updated to use Jetpack_Options 👍

Reply:
For Creative Mail, this was used by the Creative Mail team to keep track of how many installs came from Jetpack. Maybe we don't need it here since this is all in-house and we have stats?

Reply:
Discussed in Slack and decided to remove it ef6af2a